Tests/AsyncAlgorithmsTests/MultiProducerSingleConsumerChannel/MultiProducerSingleConsumerChannelTests.swift

```diff
@@ -0,0 +1,1041 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Async Algorithms open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import AsyncAlgorithms
+import XCTest
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+final class MultiProducerSingleConsumerChannelTests: XCTestCase {
+    // MARK: - sequenceDeinitialized
+
+    // Following tests are disabled since the channel is not getting deinited due to a known bug
```

Comment:
This might be a requirement to landing; id rather not introduce a type with a known leak

Reply:
I added significantly more tests now and also re-enabled those.